server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
I believe the implementation in [`_delete_worker()`](https://github.com/bmbouter/pulp/blob/6782dfe8a87593dd5579f0bc4a87dea00f4235ea/server/pulp/server/async/tasks.py#L258-L269) is race condition free. By calling `delete()` on a queryset causes both the query and delete operation to occur atomically in the database. Since it's atomic it will either delete the right thing if it was there or there is nothing to delete.